### ISSUE TITLE
[SYCL] Performance difference between Intel's OneAPI build and mine

### ISSUE THREAD

**Issue by Dr-Noob** [intel/llvm#2696]
### The issue
I installed OneAPI version `2021.1-beta09` using the package manager as explained [here](https://software.intel.com/content/www/us/en/develop/articles/installing-intel-oneapi-toolkits-via-yum.html).

I built the code in this repository using instructions from [here](https://github.com/intel/llvm/blob/sycl/sycl/doc/GetStartedGuide.md). I built it using the `--cuda` flag since I am interested in CUDA support.

I build my application with both libraries and I can see a very big performance mismatch between them. From my understanding, I should observe similar performance. When I run the code using `2021.1-beta09` library, I can see how my application exploits parallelism (it clearly launch threads), but if I run the code using my built version, the code runs in serial.

Is there any explanation for this issue? How can I solve it?

### How I build and run the code
- `2021.1-beta09`:
```
[drnoob@machine oneapi]$ head Makefile
#CXX = clang++
CXX = dpcpp

CXXFLAGS = -fsycl
LDFLAGS = -lpthread 

[drnoob@machine oneapi]$ source /opt/intel/oneapi/setvars.sh
[drnoob@machine oneapi]$ make
[drnoob@machine oneapi]$ ./exec
Using device: 'Intel(R) Xeon(R) Gold ...'
Elapsed time: 0.166s
```

- `My build`:
```
[drnoob@machine oneapi]$ head Makefile
CXX = clang++
#CXX = dpcpp

CXXFLAGS = -fsycl
LDFLAGS = -lpthread

[drnoob@machine oneapi]$ export DPCPP_HOME=~/sycl_workspace
[drnoob@machine oneapi]$ export PATH=$DPCPP_HOME/llvm/build/bin:$PATH
[drnoob@machine oneapi]$ export LD_LIBRARY_PATH=$DPCPP_HOME/llvm/build/lib:$LD_LIBRARY_PATH
[drnoob@machine oneapi]$ make
[drnoob@machine oneapi]$ ./exec
Using device: 'SYCL host device'
Elapsed time: 0.815s
```

**Comment by bader** [comment]
I don't know how `exec` chooses devices, but I assume different devices are used in these two runs.
beta09 installer configures OpenCL drivers automatically, and it seems that OpenCL CPU device is used.
`exec` built with open source compiler uses host device, which executes SYCL kernels in a single thread today.

Please, make sure that OpenCL drivers are installed according to [these instructions](https://github.com/intel/llvm/blob/sycl/sycl/doc/GetStartedGuide.md#install-low-level-runtime) for "My build" environment and `exec` chooses the same device.

**Comment by Dr-Noob** [comment]
That was it.

I skipped the `Install low level runtime` section in the instructions because I thought they were not necessary. For me, the title it's a bit misleading. 

Without following these instructions, I ran `exec` as I explained before and I got the `Using device: 'SYCL host device'` message. I have followed them now, and I get the `Using device: 'Intel(R) Xeon(R) Gold ...'` message too. The performance issues disappeared. I can confirm that now my build exploits the parallelism as Intel's OneAPI build does. I still find a small difference in performance, but I understand that it might exist such small differences.

Closing the issue.
